analysis_options.yaml

```diff
@@ -81,11 +81,11 @@ linter:
     - avoid_relative_lib_imports
     - avoid_renaming_method_parameters
     - avoid_return_types_on_setters
-    # - avoid_returning_null # there are plenty of valid reasons to return null
-    # - avoid_returning_null_for_future # not yet tested
+    # - avoid_returning_null # still violated by some pre-nnbd code that we haven't yet migrated
+    - avoid_returning_null_for_future
     - avoid_returning_null_for_void
     # - avoid_returning_this # there are plenty of valid reasons to return this
-    # - avoid_setters_without_getters # not yet tested
+    - avoid_setters_without_getters
```

Comment:
I don't fully understand the value of this, especially if it means adding unused getters to the code. The documentation is: https://dart-lang.github.io/linter/lints/avoid_setters_without_getters.html

> Defining a setter without defining a corresponding getter can lead to logical inconsistencies. Doing this could allow you to set a property to some value, but then upon observing the property's value, it could easily be different.

With the example:

```
class Bad {
  int l, r;

  set length(int newLength) {
    r = l + newLength;
  }
}
```

In this example, the only way of "observing the property's value" is by accessing the value of `r`. It seems like a setter that really just assigns a value to a public property is the anti-pattern here.

Reply:
In general, it's intuitive that something that is assignable should be readable. As in:

```dart
   foo.bar = 0;
   print(foo.bar);
```

It's very unintuitive when that kind of code fails.